dGame/dComponents/PropertyEntranceComponent.cpp

```diff
@@ -24,6 +25,15 @@ PropertyEntranceComponent::PropertyEntranceComponent(uint32_t componentID, Entit
 
 void PropertyEntranceComponent::OnUse(Entity* entity)
 {
+	auto* characterComponent = entity->GetComponent<CharacterComponent>();
+	auto* inventoryComponent = entity->GetComponent<InventoryComponent>();
+
+	if (!characterComponent || !inventoryComponent) return;
+
+	// Make sure we have a valid rocket
+	auto* rocket = inventoryComponent->FindItemById(characterComponent->GetLastRocketItemID());
```

Comment:
what if the player has binned all their rockets and made a new one?  Will this default to finding a new rocket?

Reply:
yes, the RocketEquip function, which gets called when using the launch component will set it.
However, I think this can be done, smarter:
Remove RocketEquip from LaunchpadComp::OnUse
Add the RocketEquip to the OnUse for LUPComp and PropertyEntranceComp, and LaunchpadComp::Launch

What I was trying to do was double equip, but the ItemComp has sanity check on Equip, so it wont do that